[AIRFLOW-1674] Pass execution_date to the target from TriggerDagRunOperator

sekikn opened the pull request:
Dear Airflow maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!


### JIRA
- [x] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "[AIRFLOW-XXX] My Airflow PR"
    - https://issues.apache.org/jira/browse/AIRFLOW-1674


### Description
- [x] Here are some details about my PR, including screenshots of any UI changes:

example_trigger_controller_dag can't pass conf parameter properly
to example_trigger_target_dag at least with SQLite backend, since
the format for execution_date is inconsistent between dag_run and
task_instance. This PR makes TriggerDagRunOperator pass 
execution_date to DAG#create_dagrun so as to fix it.


### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

CoreTest.test_trigger_dagrun_with_same_execution_date


### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

bolkedebruin commented on airflow/operators/dagrun_operator.py:
```diff
@@ -67,6 +67,7 @@ def execute(self, context):
             dr = trigger_dag.create_dagrun(
                 run_id=dro.run_id,
                 state=State.RUNNING,
+                execution_date=context['execution_date'],
```
This changes the default from func.now() to execution_date;

@aoen @saguziel what do you guys think?

aoen replied:
We don't use this operator at Airbnb, so take this with a grain of salt, but I could see this potentially breaking some use cases (e.g. retrying this task no longer creates a new dagrun). Also might be worth waiting for 2.0 if it breaks backwards compatibility significantly.

sekikn replied:
Thanks for the review @bolkedebruin @aoen. Releasing it as a part of 2.0 is OK to me.

sekikn committed:
[AIRFLOW-1674] Pass execution_date to the target from TriggerDagRunOperator

example_trigger_controller_dag can't pass conf parameter properly
to example_trigger_target_dag at least with SQLite backend, since
the format for execution_date is inconsistent between dag_run and
task_instance. This PR makes TriggerDagRunOperator pass
execution_date to DAG#create_dagrun so as to fix it.
